persist/fs/retriever.go

```diff
@@ -101,6 +106,10 @@ func NewBlockRetriever(
 		bytesPool:      opts.BytesPool(),
 		status:         blockRetrieverNotOpen,
 		notifyFetch:    make(chan struct{}, 1),
+		// We just close this channel when the fetchLoops should shutdown, so no
+		// buffering is required
+		fetchLoopsShouldShutdownCh: make(chan struct{}),
+		fetchLoopsHaveShutdownCh:   make(chan struct{}, opts.FetchConcurrency()),
```

Comment:
If you're just waiting for this channel to be shut then it doesn't need to be of a specific size.